GI/GLib/Functions.hs:5085:34: error:     • Couldn't match type ‘Char’ with ‘CInt’

I dunno what changed but gi-glib-2.0.27 stopped building in Fedora 38+ (devel):

```
[102 of 120] Compiling GI.GLib.Functions

GI/GLib/Functions.hs:5085:34: error:
    • Couldn't match type ‘Char’ with ‘CInt’
      Expected: Ptr CInt
        Actual: Ptr Char
    • In the first argument of ‘g_unicode_canonical_ordering’, namely
        ‘string'’
      In a stmt of a 'do' block: g_unicode_canonical_ordering string' len
      In the second argument of ‘($)’, namely
        ‘do let len = fromIntegral $ length string
            string' <- packStorableArray string
            g_unicode_canonical_ordering string' len
            freeMem string'
            ....’
     |
5085 |     g_unicode_canonical_ordering string' len
     |                                  ^^^^^^^
```
This is with ghc-9.2.6.

Perhaps I should try with ghc-9.2.7 too?

--- a/lib/Data/GI/CodeGen/Conversions.hsc
+++ b/lib/Data/GI/CodeGen/Conversions.hsc
@@ -306,6 +306,8 @@ hToF' t a hType fType transfer
         return $ M "(packMapStorableArray realToFrac)"
     | TCArray False _ _ (TBasicType TDouble) <- t =
         return $ M "(packMapStorableArray realToFrac)"
+    | TCArray False _ _ (TBasicType TUniChar) <- t =
+        return $ M "(packMapStorableArray ((fromIntegral . ord)))"
     | TCArray False _ _ (TBasicType _) <- t =
         return $ M "packStorableArray"
     | TCArray False _ _ TGValue <- t =

